doc/src/manual/types.md

```diff
@@ -973,6 +966,39 @@ julia> Ptr{Int64} <: Ptr
 true
 ```
 
+## UnionAll Types
+
+We have said that a parametric type like `Ptr` acts as a supertype of all its instances
+(`Ptr{Int64}` etc.). How does this work? `Ptr` itself cannot be a normal data type, since without
+knowing the type of the referenced data the type clearly cannot be used for memory operations.
+The answer is that `Ptr` (or other parametric type like `Array`) is a different kind of type called a
+`UnionAll` type. Such a type expresses the *iterated union* of types for all values of some variable.
+
+`UnionAll` types are usually written using the keyword `where`. For example `Ptr` could be more
+accurately written as `Ptr{T} where T`, meaning all values whose type is `Ptr{T}` for some value
+of `T`. Each `where` introduces a single variable, so these expressions are nested for types with
```

Comment:
"Variable" is the only appropriate word I can think of here. It refers to a placeholder that might have various different values.